homeassistant/components/decora_wifi/common.py

```diff
@@ -0,0 +1,203 @@
+"""Common code for decora_wifi."""
+
+from __future__ import annotations
+
+from collections.abc import Callable
+import logging
+
+from decora_wifi import DecoraWiFiSession
+from decora_wifi.models.iot_switch import IotSwitch
+from decora_wifi.models.person import Person
+from decora_wifi.models.residence import Residence
+from decora_wifi.models.residential_account import ResidentialAccount
+from decora_wifi.models.residential_permission import ResidentialPermission
+
+from homeassistant.const import EVENT_HOMEASSISTANT_STOP
+from homeassistant.core import HomeAssistant
+from homeassistant.exceptions import HomeAssistantError
+from homeassistant.helpers import device_registry
+from homeassistant.helpers.entity import DeviceInfo, Entity
+
+from .const import LIGHT_DOMAIN, PLATFORMS
+
+_LOGGER = logging.getLogger(__name__)
+
+
+class DecoraWifiError(HomeAssistantError):
+    """Base for errors raised when the decora_wifi integration encounters an issue."""
+
+
+class SessionEntityNotFound(DecoraWifiError):
+    """Raised if a platform fails to find the session entity during setup."""
+
+
+class LoginFailed(DecoraWifiError):
+    """Raised when DecoraWifiPlatform.login() fails to log in."""
+
+
+class CommFailed(DecoraWifiError):
```

Comment:
Can you move these exception into the library instead ?

Reply:
Unfortunately, no.  I'm not a maintaner over there and [the PR I created back in September](https://github.com/tlyakhov/python-decora_wifi/pull/19) for the purpose of having the library raise named exceptions has received no comment or communication at all from @tlyakhov.  Further progress on this item would probably require a fork.

Reply:
It looks like his email is on his github: tlyakhov@gmail.com

If after reaching out he doesn't respond in a few days, given the lack of github activity https://github.com/tlyakhov , a fork seems fine.

Reply:
Apologies folks, just saw this thread. A fork is probably a good idea, I no longer even have a Decora WiFi device to test things on. I did merge the PR referenced but I don't even have a dev environment set up at the moment.